src/pygama/lgdo/table.py

```diff
@@ -243,33 +247,70 @@ def eval(self, expr_config: dict) -> Table:
-              :meth:`pandas.DataFrame.eval` (see also `here
-              <https://pandas.pydata.org/pandas-docs/stable/user_guide/enhancingperf.html#expression-evaluation-via-eval>`_
-              for documentation).
+              :meth:`numexpr.evaluate` (see also `here
+              <https://numexpr.readthedocs.io/projects/NumExpr3/en/latest/index.html>`_
+              for documentation). Note: because of internal limitations, reduction operations must appear the last in the stack.
             - ``parameters`` is a dictionary of function parameters. Passed to
-              :meth:`pandas.DataFrame.eval` as `local_dict` argument.
+              :meth:`numexpr.evaluate`` as `local_dict` argument.
 
 
         Warning
         -------
         Blocks in `expr_config` must be ordered according to mutual dependency.
         """
-        df = self.get_dataframe()
         out_tbl = Table(size=self.size)
-
-        # evaluate expressions one-by-one (in order) to make sure expression
-        # dependencies are satisfied
         for out_var, spec in expr_config.items():
-            df.eval(
-                f"{out_var} = {spec['expression']}",
-                parser="pandas",
-                engine="numexpr",  # this should be faster than Python's native eval() for n_rows > 1E4, see Pandas docs
-                local_dict=spec["parameters"] if "parameters" in spec else None,
-                inplace=True,
-            )
-
-            # add column to output LGDO Table
-            out_tbl.add_column(out_var, Array(df[out_var].to_numpy()))
+            in_vars = {}
+            # Find all valid python variables in expression (e.g "a*b+sin(Cool)" --> ['a','b','sin','Cool'])
+            for elem in re.findall(r"\s*[A-Za-z_]\w*\s*", spec["expression"]):
+                elem = elem.strip()
+                if elem in self:  # check if the variable comes from dsp
+                    in_vars[elem] = self[elem]
+                elif (
+                    elem in out_tbl.keys()
+                ):  # if not try from previously processed data, else ignore since it is e.g sin func
+                    in_vars[elem] = out_tbl[elem]
+
+                else:
+                    continue
+                # get the nda if it is an Array instance
+                if isinstance(in_vars[elem], Array):
+                    in_vars[elem] = in_vars[elem].nda
+                # No vector of vectors support yet
+                elif isinstance(in_vars[elem], VectorOfVectors):
+                    raise TypeError("Data of type VectorOfVectors not supported (yet)")
+
+            out_data = ne.evaluate(
+                f"{spec['expression']}",
+                local_dict=dict(in_vars, **spec["parameters"])
+                if "parameters" in spec
+                else in_vars,
+                global_dict=None,
+                optimization="moderate",  # Slow, but calculation is accurate (alternative "aggressive")
+                truediv="auto",
```

Comment:
I would remove these (i.e. default behavior), I don't think we care. My intuition is that the default `optimization='aggressive'` is fine, we are not doing precision calculations (or non-trivial floating-point operations).

Reply:
Done
i.e optimization is now 'aggressive' (default option) --> faster but small errors can occur